### ISSUE TITLE
Checkbox: label only clickable when simple element

### ISSUE THREAD

**Issue by agriffis** [Semantic-Org/Semantic-UI-React#3683]
## Bug Report
Checkbox label only clickable when simple element

![Sandbox-CodeSandbox](https://user-images.githubusercontent.com/50637/60183886-1cf34f00-97f5-11e9-933e-ef2c5e796831.gif)

### Steps
Nest a span in label, and clicking no longer works

### Expected Result
Toggle the checkbox

### Actual Result
Nothing

### Version
0.87.2

### Testcase
https://codesandbox.io/s/semantic-ui-react-78ovl

I haven't verified this, but my guess is this from #3435 which requires an exact match for the click event on the label element, and doesn't account for clicks on nested elements.

**Comment by agriffis** [comment]
Here's the workaround I'm using:

```jsx
const ClickableLabel = props => {
  const label = React.createRef()
  const handleClick = e => {
    if (e.target !== label.current) {
      label.current.click()
      return false
    }
  }
  return <label {...props} onClick={handleClick} ref={label} />
}
```

**Comment by layershifter** [comment]
Fixed in #3692, thanks for the report 👍
